docs/tutorials/using_apis/connectivity/mesh_tutorial.md

```diff
@@ -0,0 +1,136 @@
+<h1 id="mesh-tutorial">Mesh tutorial</h1>
```

Comment:
I like parts of this tutorial a lot, but it's lacking some cohesion. What is the main thing this tutorial is trying to accomplish? What does it teach you how to do?

Reply:
This tutorial should answer to question "What mesh protocol to select". It looks like we should include also https://jira.arm.com/browse/IOTTHD-3368 here as it might provide the missing glue.